Foundation/NSNotificationQueue.swift

```diff
@@ -15,26 +23,116 @@ public struct NSNotificationCoalescing : OptionSetType {
 }
 
 public class NSNotificationQueue : NSObject {
-    
+
+    internal typealias NotificationQueueList = Array<NSNotificationQueue>
+    internal typealias NSNotificationListEntry = (NSNotification, [String]) // Notification ans list of modes the notification may be posted in.
+    internal typealias NSNotificationList = [NSNotificationListEntry] // The list of notifications to post
+
+    internal let notificationCenter: NSNotificationCenter
+    internal var asapList = NSNotificationList()
+    internal var idleList = NSNotificationList()
+
+    // The NSNotificationQueue instance is associated with current thread.
+    // The _notificationQueueList represents a list of notification queues related to the current thread.
+    private static var _notificationQueueList = NSThreadSpecific<NSArray>()
+    internal class var notificationQueueList: NotificationQueueList {
+        get {
+            let array = _notificationQueueList.get({
+                return NSArray()
+            })
+            return array.allObjects as! NotificationQueueList
```

Comment:
err disregard, I misread the type aliasing; it still crashes the compiler on linux

This will do roughly the same thing except it won't crash the compiler.
```swift
internal static var notificationQueueList: NotificationQueueList {
        get {
            let array = _notificationQueueList.get() {
                return NSArray()
            }
	    return array._swiftObject.map { $0 as! NSNotificationQueue }
        }
        set {
            _notificationQueueList.set(newValue._nsObject)
        }
    }
```